zowe/zowe-cli
# [v3] Download special char issue

What It Does
This PR resolves an issue where special characters could become corrupted when downloading large files. The corruption was caused by each chunk buffer being converted to a string when calling the processNewLines() function and then converted back to a buffer. This PR introduces support for passing buffers directly into the processNewLines() function to prevent corruption.

How to Test
The issue arises when a special character appears as the last character of a stream chunk. To reproduce, create a PDS file containing exclusively special characters with a size larger than the default chunk size (>8KB). In such cases, the last character of each chunk may be corrupted and appear as two unknown characters.

To observe this issue, download a PDS file that meets the criteria using the following example command (assuming the use of IBM-1147 encoding):

`zowe zos-file download ds <PDSMEMBER> --encoding "IBM-1147"`

where `<PDSMEMBER> `is a file larger than 8KB containing special characters.

Example Line (about 105 are required in a file to exceed 8KB):
`àèéìòùÀÈÉÌÒÙ°àèéìòùÀÈÉÌÒÙ°àèéìòùÀÈÉÌÒÙ°àèéìòùÀÈÉÌÒÙ°àèéìòùÀÈÉÌÒÙ°àèéìòùÀÈÉÌÒÙ°à`

**Review Checklist**
I certify that I have:
